wp-content/plugins/goonj-blocks/src/render.php

```diff
@@ -1,88 +1,112 @@
-				<td class="wp-block-gb-table-cell wp-block-gb-table-header">From</td>
-				<td class="wp-block-gb-table-cell"><?php echo $start_date->format( 'd-m-Y h:i A' ); ?></td>
-			</tr>
-			<tr class="wp-block-gb-table-row">
-				<td class="wp-block-gb-table-cell wp-block-gb-table-header">To</td>
-				<td class="wp-block-gb-table-cell"><?php echo $end_date->format( 'd-m-Y h:i A' ); ?></td>
-			</tr>
-			<tr class="wp-block-gb-table-row">
-				<td class="wp-block-gb-table-cell wp-block-gb-table-header">Address of the camp</td>
-				<td class="wp-block-gb-table-cell"><?php echo esc_html( $address ); ?></td>
-			</tr>
-		</tbody>
-	</table>
-	<div <?php echo get_block_wrapper_attributes(); ?>>
-		<a href="<?php echo esc_url( $register_link ); ?>" class="wp-block-gb-action-button">
-			<?php esc_html_e( 'Volunteer with Goonj', 'goonj-blocks' ); ?>
-		</a>
-		<a href="<?php echo esc_url( $material_contribution_link ); ?>" class="wp-block-gb-action-button">
-			<?php esc_html_e( 'Record your Material Contribution', 'goonj-blocks' ); ?>
-		</a>
-	</div>
-	<?php elseif ( 'processing-center' === $target ) : ?>
-		<table class="wp-block-gb-table">
-			<tbody>
-				<tr class="wp-block-gb-table-row">
-					<td class="wp-block-gb-table-cell wp-block-gb-table-header">Address</td>
-					<td class="wp-block-gb-table-cell"><?php echo CRM_Utils_Address::format( $action_target['address'] ); ?></td>
-				</tr>
-			</tbody>
-		</table>
-		<div <?php echo get_block_wrapper_attributes(); ?>>
-			<a href="<?php echo esc_url( $pu_visit_check_link ); ?>" class="wp-block-gb-action-button">
-				<?php esc_html_e( 'Office Visit', 'goonj-blocks' ); ?>
-			</a>
-			<a href="<?php echo esc_url( $pu_material_contribution_check_link ); ?>" class="wp-block-gb-action-button">
-				<?php esc_html_e( 'Material Contribution', 'goonj-blocks' ); ?>
-			</a>
-		</div>
-<?php endif;
+if (in_array($target, ['collection-camp', 'dropping-center'])) :
+  try {
+    if ('dropping-center' === $target) {
+      $start_date        = new DateTime($action_target['Dropping_Centre.Start_Time']);
+      $end_date          = new DateTime($action_target['Dropping_Centre.End_Time']);
+      $address           = $action_target['Dropping_Centre.Where_do_you_wish_to_open_dropping_center_Address_'];
+      $contribution_link = $dropping_center_material_contribution_link;
+    }
+    else {
+      $start_date        = new DateTime($action_target['Collection_Camp_Intent_Details.Start_Date']);
+      $end_date          = new DateTime($action_target['Collection_Camp_Intent_Details.End_Date']);
+      $address           = $action_target['Collection_Camp_Intent_Details.Location_Area_of_camp'];
+      $contribution_link = $material_contribution_link;
+    }
+  }
+  catch (Exception $e) {
+    // Log the error and set fallback dates.
+    \Civi::log()->info('Invalid date format for start or end time', ['error' => $e->getMessage()]);
+    $start_date = new DateTime();
+    $end_date   = new DateTime();
+  }
```

Comment:
but, this error handling is essential. If the start date or end date format is invalid, it will catch the error and log it, making it easier to debug if this issues arise in the future. If you have any further questions about this, feel free to ask.